spec/models/sms/decoder_spec.rb

```diff
@@ -66,14 +66,29 @@
       expect_decoding_fail(nil, body: "abc 1.15", error: "form_not_found")
     end
 
-    it "submitting to outdated form should produce appropriate error" do
-      form = create_form(questions: %w[integer])
+    context "with multiple versions" do
+      let(:form) { create_form(questions: %w[integer]) }
+      let(:v1) { form.versions.first }
+      let(:v2) { form.versions.second }
+      let(:v3) { form.versions.third }
+
+      before do
+        2.times { form.upgrade_version! }
+        v1.update!(is_oldest_accepted: false)
+        v2.update!(is_oldest_accepted: true)
+      end
+
+      it "outdated version should produce appropriate error" do
+        expect_decoding_fail(form, body: "#{v1.code} 1.15", error: "form_version_outdated")
+      end
 
-      # upgrade form version before submitting
-      old_version_code = form.current_version.code
-      form.upgrade_version!
+      it "current accepted version should be accepted" do
```

Comment:
This is a bit confusing since v2 is not the current version. Perhaps "minimum version should be accepted" ;)